analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EqualityOnFloatingPoint.cs

```diff
@@ -2,86 +2,132 @@
 using System.Collections.Generic;
 using System.Runtime.Versioning;
 
-namespace Tests.Diagnostics
+public class EqualityOnFloatingPoint
 {
-    public class EqualityOnFloatingPoint
+    void Test(float f, double d1, double d2)
     {
-        void test(float f, double d1, double d2)
+        dynamic din = null;
+        if (din == null) { }
```

Comment:
Do we really need this? It doesn't seem to test anything relevant. If we want to keep it:

* Remove the local declaration and make it a parameter instead
* Rename the variable to `dyn`.
* Change the test case so that a floating point number is involved on one side of the binary.

Reply:
Indeed, we don't need this check in the way it currently is in the code.

However, applying your suggestions:

```cs
void Test(float f, double d1, double d2, dynamic dyn)
    {
        if (dyn == 3.14) { }        // Noncompliant {{Do not check floating point equality with exact values, use a range instead.}}
```

It becomes a false negative, as in the following example (which prints `False`):
```cs
dynamic x = 0.1 + 0.2;
var b = x == 0.3;
Console.WriteLine(b);
```

The problem comes from:
```cs
context.SemanticModel.GetSymbolInfo(equals).Symbol is IMethodSymbol { ContainingType: { } container, Name: { } equalitySymbolName }
```
where `Name` is correctly reported as `op_Equality`, but `ContainingType` is not (`null`).

If you agree, I would leave this as a FP, and not try to infer the true type of equality in presence of `dynamic`.